tests/drivers/gpio/gpio_nrf/testcase.yaml

```diff
@@ -0,0 +1,10 @@
+common:
+  tags: drivers gpio
+  depends_on: gpio
+  harness: ztest
+tests:
+  drivers.gpio.gpio_nrf:
+    filter: dt_compat_enabled("nordic,nrf-gpio")
```

Comment:
maybe you could add `and dt_nodelabel_enabled("led0")` and remove `platform_exclude`.